bazaarvoice/bv-ui-core
# Adding messageFormat module for INTL API

# PR description



## JIRA tickets

[PD-191705](https://bazaarvoice.atlassian.net/browse/PD-191705)



## This PR is for

- Feature 



## Requirements / issues

- We have a functionality to convert locale to a country name. 
- For this we are using INTL.DisplayNames. Since this function will be used in multiple places and it is using the browser API . We can have the code in bv-ui-core



## Solutions

- Using the INTL.DisplayNames API . We can convert the locale  to a country. 
-



## Unit tests

- Added messageFormat unit test cases 
- Added Chrome Headless as a browser to run unit test cases instead of PhantomJS as INTL was not available in PhantomJS
- Fixed Test cases for performance module 



## Steps to verify

- Copy paste the function in the browser and check if it is performing as expected 
- 
<img width="821" alt="Screenshot 2023-03-29 at 12 01 06 PM" src="https://user-images.githubusercontent.com/78583970/228445872-d0eb85de-710b-479b-847c-34d590c915f1.png">


## Check points



- [x] [Design / approach [doc](https://bazaarvoice.atlassian.net/wiki/spaces/DEV/pages/78216462383/CFR+21+-+Display+Native+reviews+Design+Doc)


- [x] Unit test cases

## Version 
This will be released as a new minor version as we are adding a new module 


[PD-191705]: https://bazaarvoice.atlassian.net/browse/PD-191705?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
